src/cliffordsynthesis/CliffordSynthesizer.cpp

```diff
@@ -170,6 +182,8 @@
     upperBound = std::min(upperBound, results.getGates());
   } else if (config.targetMetric == TargetMetric::Depth) {
     upperBound = std::min(upperBound, results.getDepth());
+  } else if (config.targetMetric == TargetMetric::TQDepth) {
+    upperBound = std::max(upperBound, results.getTQDepth());
```

Comment:
As we have discussed, this can't be right. The optimal synthesis result cannot have a higher depth then either of these values.